python/langsmith/_expect.py

```diff
@@ -72,7 +77,7 @@ def __init__(
             max_workers=3
         )
         rt = rh.get_current_run_tree()
-        self._run_id = rt.id if rt else run_id
+        self._run_id = rt.trace_id if rt else run_id
```

Comment:
If you're running this within a nested function, we'd prefer it by default to be assigned to the root trace (since it's called beneath `@unit`